bpf/proxy/syncer.go

```diff
@@ -271,6 +273,16 @@ func (s *Syncer) cleanupDerived(id uint32) error {
 			if err := s.bpfSvcs.Delete(key[:]); err != nil {
 				return errors.Errorf("bpfSvcs.Delete: %s", err)
 			}
+			keys, err := getSvcNATKeyLBSrcRange(si.svc)
+			if err != nil {
+				return err
+			}
+			for _, key = range keys {
+				log.Debugf("bpf map deleting derived %s:%s", key, nat.NewNATValue(id, 0, 0, 0))
+				if err := s.bpfSvcs.Delete(key[:]); err != nil {
+					return errors.Errorf("bpfSvcs.Delete: %s", err)
+				}
+			}
```

Comment:
I think this is executed for every occurrence of the ID in the prevMap, that is for the ClusterIP and ExternalIP as well - the LB is unlikely to have any other entries. You could do it just for the ExternalIP. No a big deal, but if `writeLBSrcRangeSvcNATKey` inserted those src range entries in the `newSvcMap` they would be in the `prevSvcMap` here.

Reply:
I am not sure if we can distinguish whether it is external IP or cluster IP here